wrappers/csharp/cs-tutorial-8-D400-on-chip-calibration/ExampleAutocalibrateDevice.cs

```diff
@@ -0,0 +1,541 @@
+                }
+                Console.WriteLine($"Report about testing calibration modes has been written to file:\n\t{fileName}");
+                Console.WriteLine("Testing calibration modes complete");
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"{Environment.NewLine} Error during calibration:{Environment.NewLine} {ex.Message}");
+            }
+
+        }
+
+        private string GetHeader4FileReport()
+        {
+            var row = new List<string>();
+            row.Add("depthProfileSet");
+            row.Add("Width");
+            row.Add("Height");
+            row.Add("Framerate");
+            row.Add("Format");
+            row.Add("calibrationSet");
+            row.Add("(int)CalibrationSpeed");
+            row.Add("(int)CalibrationScanParameter");
+            row.Add("(int)CalibrationDataSampling");
+            row.Add("CalibrationSpeed");
+            row.Add("CalibrationScanParameter");
+            row.Add("CalibrationDataSampling");
+            //row.Add("timeout (min:sec.millisec)");
+            row.Add("calibrationTakes (min:sec.millisec)");
+            row.Add("Device health");
+            row.Add("Device health description");
+            row.Add("Comments");
+
+            return string.Join(",", row);
+        }
+        private string GetRow4FileReport(
+            int depthProfileSet,
+            VideoStreamProfile profile,
+            int calibrationSet,
+            CalibrationSpeed calibrationSpeed,
+            CalibrationScanParameter calibrationScanParameter,
+            CalibrationDataSampling calibrationDataSampling,
+            int timeout,
+            TimeSpan calibrationTakes,
+            string deviceHealth,
+            string deviceHealthDescription,
+            string comments = ""
+            )
+        {
+            var row = new List<string>();
+            row.Add($"{depthProfileSet}");
+            row.Add($"{profile.Width}");
+            row.Add($"{profile.Height}");
+            row.Add($"{profile.Framerate}");
+            row.Add($"{profile.Format}");
+            row.Add($"{calibrationSet}");
+            row.Add($"{(int)calibrationSpeed}");
+            row.Add($"{(int)calibrationScanParameter}");
+            row.Add($"{(int)calibrationDataSampling}");
+            row.Add($"{calibrationSpeed}");
+            row.Add($"{calibrationScanParameter}");
+            row.Add($"{calibrationDataSampling}");
+            //row.Add($@"{timeout:mm\:ss\.fff}"); 
+            row.Add($@"{calibrationTakes:mm\:ss\.fff}");
+            row.Add($"{deviceHealth}");
+            row.Add($"{deviceHealthDescription}");
+            row.Add($"{comments}");
+
+            return string.Join(",", row);
+        }
+        private static bool IsLinux
+        {
+            get
+            {
+                int p = (int)Environment.OSVersion.Platform;
+                return (p == 4) || (p == 6) || (p == 128);
```

Comment:
Please add comment to explain these values

Reply:
Link added